sdks/java/extensions/sql/datacatalog/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/datacatalog/DataCatalogTableProvider.java

```diff
@@ -98,9 +98,8 @@ public void dropTable(String tableName) {
   }
 
   @Override
-  public @Nullable Table getTable(String tableNamePart) {
-    throw new UnsupportedOperationException(
-        "Loading a table by partial name '" + tableNamePart + "' is unsupported");
+  public @Nullable Table getTable(String tableName) {
+    return loadTable(tableName);
```

Comment:
This should always fail to load, since it will only be called with the last component of a table name. If this works, then I think the problem is that the ZetaSQLPlanner needs to insert the components of the SQL name, splitting them, instead of one string.

Reply:
I dug through the code, and I think what is required is to replace "JOIN_INTO_COMPOUND_ID" with identity function here: https://github.com/apache/beam/blob/master/sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/TableResolutionContext.java#L44

Reply:
I think your understanding is wrong. Apparently it is the whole string passed in to called load table.

Reply:
My last comment was to address the first comment. 

Regarding to the second comment. @kennknowles, if you know how to fix this problem, can you send a PR?

Reply:
I think this function was correct before and is now wrong.

Reply:
Agreed.

Reply:
This line should not be changed.